packages/backend/src/open_payments/quote/service.ts

```diff
@@ -185,10 +189,20 @@ async function createQuote(
         receiver,
         receiveAmount: options.receiveAmount,
         debitAmount: options.debitAmount
+          ? {
+              ...options.debitAmount,
+              value: options.debitAmount.value - fixedDebitFee
+            }
+          : undefined
       }
     )
     stopTimerQuote()
 
+    // Calculate fee when receiveAmount is set
+    const fixedReceiveFee = !options.debitAmount
+      ? sendingFee?.calculate(quote.debitAmount.value) ?? 0n
+      : 0n
```

Comment:
```suggestion
    const fixedReceiveFee = options.receiveAmount
      ? sendingFee?.calculate(quote.debitAmount.value) ?? 0n
      : 0n
```

Reply:
Just to match the comment better

Reply:
It would also be good to add just a bit in the comment why we need to apply this fee _after_ we got the quote.